apps/emqx_dashboard/test/emqx_dashboard_https_SUITE.erl

```diff
@@ -44,6 +45,45 @@ end_per_suite(_Config) -> emqx_mgmt_api_test_util:end_suite([emqx_management]).
 init_per_testcase(_TestCase, Config) -> Config.
 end_per_testcase(_TestCase, _Config) -> emqx_mgmt_api_test_util:end_suite([emqx_management]).
 
+t_update_conf(_Config) ->
+    Conf = #{
+        dashboard => #{
+            listeners => #{
+                https => #{bind => 18084, enable => true},
+                http => #{bind => 18083, enable => true}
+            }
+        }
+    },
+    emqx_common_test_helpers:load_config(emqx_dashboard_schema, Conf),
+    emqx_mgmt_api_test_util:init_suite([emqx_management], fun(X) -> X end),
+    Headers = emqx_dashboard_SUITE:auth_header_(),
+    {ok, Client1} = emqx_dashboard_SUITE:request_dashboard(
+        get, https_api_path(["clients"]), Headers
+    ),
+    {ok, Client2} = emqx_dashboard_SUITE:request_dashboard(
+        get, http_api_path(["clients"]), Headers
+    ),
+    ?assertEqual(Client1, Client2),
+    Raw = emqx:get_raw_config([<<"dashboard">>]),
+    Raw1 = emqx_utils_maps:deep_put([<<"listeners">>, <<"https">>, <<"enable">>], Raw, false),
+    ?assertMatch({ok, _}, emqx:update_config([<<"dashboard">>], Raw1)),
+    {ok, Client3} = emqx_dashboard_SUITE:request_dashboard(
+        get, http_api_path(["clients"]), Headers
+    ),
+    ?assertEqual(Client1, Client3),
+    ?assertMatch(
+        {ok, _}, emqx_dashboard_SUITE:request_dashboard(get, https_api_path(["clients"]), Headers)
```

Comment:
Q: shouldn't this fail, since we've just disabled the https listener?

Reply:
Good catch, updated